### PR TITLE
Add "insert" function to the CameraFrontEnd like in FlxGroups

### PR DESCRIPTION
**Pull request by T5mpler**
Simple pull request that adds an insert function to the CameraFrontEnd class incase you want a camera to render in-between another camera in the list (I'm surprised this didn't exist already unless there's a reason).

### CODE REVIEW

**Comment by T5mpler** [comment]
Billezen back again 😔

**Comment by system32unknown** [comment]
hi t5

File: flixel/system/frontEnds/CameraFrontEnd.hx [hunk 1]
```diff
@@ -74,6 +74,34 @@ class CameraFrontEnd
...
+	/**
+	 * Inserts a new camera object to the game.
+	 *
+	 * @param	NewCamera         The camera you want to add.
+	 * @param	Position          The position in the list where you want to insert the camera
+	 * @param	DefaultDrawTarget Whether to add the camera to the list of default draw targets. If false, 
+	 *                            `FlxBasics` will not render to it unless you add it to their `cameras` list.
+	 * @return	This FlxCamera instance.
+	 */
+	public function insert<T:FlxCamera>(NewCamera:T, Position:Int, DefaultDrawTarget:Bool = true):T
```
Comment (Geokureli) [inline flixel/system/frontEnds/CameraFrontEnd.hx:87]: The official rule is to use lowercase arg names, we are slowly changing old uppercase args as we  go

File: flixel/system/frontEnds/CameraFrontEnd.hx [hunk 2]
```diff
@@ -74,6 +74,34 @@ class CameraFrontEnd
...
+		if (DefaultDrawTarget)
+			defaults.push(NewCamera);
+		
+		for (i in 0...list.length)
+		{
+			list[i].ID = i;
+		}
+		cameraAdded.dispatch(NewCamera);
+		return NewCamera;
+	}
```
Comment (Geokureli) [inline flixel/system/frontEnds/CameraFrontEnd.hx:104]:
```suggestion
	/**
	 * Inserts a new camera object to the game.
	 * 
	 * - If `position` is negative, `list.length + position` is used
	 * - If `position` exceeds `list.length`, the camera is added to the end.
	 *
	 * @param	newCamera         The camera you want to add.
	 * @param	position          The position in the list where you want to insert the camera
	 * @param	defaultDrawTarget Whether to add the camera to the list of default draw targets. If false, 
	 *                            `FlxBasics` will not render to it unless you add it to their `cameras` list.
	 * @return	This FlxCamera instance.
	 */
	public function insert<T:FlxCamera>(newCamera:T, position:Int, defaultDrawTarget = true):T
	{
	    // negative numbers are relative to the length (match Array.insert's behavior)
	    if (position < 0)
	        position += list.length;
	    
	    // invalid ranges are added (match Array.insert's behavior)
        if (position >= list.length)
            return add(newCamera);
        
        final childIndex = FlxG.game.getChildIndex(list[position]);
        FlxG.game.addChildAt(newCamera.flashSprite, childIndex);
		
		list.insert(position, newCamera);
		if (defaultDrawTarget)
			defaults.push(newCamera);
		
		for (i in position...list.length)
			list[i].ID = i;
		
		cameraAdded.dispatch(newCamera);
		return newCamera;
	}
```

**Review by Geokureli — CHANGES_REQUESTED** [review]
Mostly style changes. I also made insert match the behavior of haxe's Array.insert

**Review by Geokureli — APPROVED** [review]

File: flixel/system/frontEnds/CameraFrontEnd.hx [hunk 3]
```diff
@@ -74,6 +74,42 @@ class CameraFrontEnd
...
+	{
+	    // negative numbers are relative to the length (match Array.insert's behavior)
+	    if (position < 0)
+	        position += list.length;
+	    
+	    // invalid ranges are added (match Array.insert's behavior)
+        if (position >= list.length)
+            return add(newCamera);
+        
+        final childIndex = FlxG.game.getChildIndex(list[position]);
```
Comment (Geokureli) [inline flixel/system/frontEnds/CameraFrontEnd.hx:100]:
```suggestion
        final childIndex = FlxG.game.getChildIndex(list[position].flashSprite);
```

**Comment by T5mpler** [comment]
Ah got it! I only did the naming and stuff like that because  I was trying to match how the other functions did it in CameraFrontEnd, much appreciations!

**Comment by Geokureli** [comment]
> Ah got it! I only did the naming and stuff like that because I was trying to match how the other functions did it in CameraFrontEnd, much appreciations!

Makes sense, thanks!

**Pull request merged**